core/server/proxy/src/main/java/alluxio/proxy/s3/tag/TagHelper.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.tag;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.exception.AlluxioException;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.proxy.s3.S3RestServiceHandler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper class for handling tags according to https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketTagging.html
+ * and https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectTagging.html.
+ */
+public class TagHelper {
+  private static final Logger LOG = LoggerFactory.getLogger(S3RestServiceHandler.class);
+
+  /**
+   * all tag information will go into TAG_FOLDER (/_TAG), inside of the folder, every file in the
```

Comment:
This folder is in the root of the alluxio filesystem? I think we have a special folder that we can use for alluxio specific things. Lu should have experience with that.

Also, if the root UFS is CACHE_THROUGH, is that ok?

Reply:
for the root UFS being CACHE_THROUGH. How is this any different from the problem with _internal/ for catalog service.

Also: @LuQQiu

Reply:
We have a `alluxio.underfs.s3.directory.suffix` to use a special zero-byte object name for the empty folder, besides that, i don't remember we have any special folder for other usage.

Reply:
O whoops, I think it was the tmp file issue that @HelloHorizon  worked on. I think there is a special folder for Alluxio to use internally.

Reply:
The tmp file folder for persistence is the folder under UFS mount point `alluxio.util.io.PathUtils#getPersistentTmpPath`. And we also have a similar implementation for block in `alluxio.worker.BlockUtils#getUfsBlockPath`.